src/Features/Core/Portable/Shared/Utilities/IOUtilities.cs

```diff
@@ -23,26 +23,21 @@ public static T PerformIO<T>(Func<T> function, T defaultValue = default(T))
             {
                 return function();
             }
-            catch (IOException)
-            {
-            }
-            catch (SecurityException)
-            {
-            }
-            catch (ArgumentException)
-            {
-            }
-            catch (UnauthorizedAccessException)
-            {
-            }
-            catch (NotSupportedException)
-            {
-            }
-            catch (InvalidOperationException)
+            catch (Exception e) when (IsNormalIOException(e))
             {
             }
 
             return defaultValue;
         }
+
+        public static bool IsNormalIOException(Exception e)
+        {
+            return e is IOException ||
+                   e is SecurityException ||
+                   e is ArgumentException ||
```

Comment:
i'm not sure htat's true when it comes to the windows FS apis.  for example, you might try to name something "con" and you could get an ArgumentException.  There is no APi that says "this name is ok", so you always run the risk of having a name that will cause an exception.